### ISSUE TITLE
Top-level variable += 1 does not work inside macro-inlined module method

### ISSUE THREAD

**Issue by r00ster91** [crystal-lang/crystal#8248]
```cr
module Module
  def method
    puts variable
  end
end

variable = 5

{{ Module.resolve.methods[0].body }}
```
```
5
```
But adding `variable += 1`:
```cr
module Module
  def method
    variable += 1
    puts variable
  end
end

variable = 5

{{ Module.resolve.methods[0].body }}
```
```
In bug.cr:3:16

 3 | variable += 1
                ^
Error: '+=' before definition of 'variable'
```
Even though `variable` is clearly already defined.
It doesn't have to be `variable += 1`, it can also be `variable = variable + 1` or any other variable expression that uses its variable in an assignment.

**Comment by r00ster91** [comment]
Hmm could be a duplicate of #7975 but not really I think because there `x` can not even be printed but here it can, just the assignments don't work.

**Comment by asterite** [comment]
variable += 1 in a method, if you didn't define variable before, is a syntax error.

My feeling toward this: if you want to inject code, use a macro. And then: macros shouldn't use variables defined outside of them.

What are you trying to do?

**Comment by r00ster91** [comment]
> variable += 1 in a method, if you didn't define variable before, is a syntax error.

Yeah sure, but in this case the method's body is inlined so it's not really in the method, but on the top level where `variable` is defined and `puts variable` even works, while only assignments don't work which doesn't make any sense.

Actually, I found a really good way now without macro-inlining anything.

I will do it like this:

```cr
Module.method({{ Module.resolve.methods[0].args.join(',').id }})
```

And at `def method(...)` in the module I just specify the variables I want to use from the outside.
So I do it the way it should be done and don't do it in a hacky macro way like I did before.
So this is solved now but the compiler bug is still there.
